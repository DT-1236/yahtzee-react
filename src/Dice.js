import React, { Component } from 'react';
import Die from './Die';
import styled from 'styled-components';

const StyledWrapper = styled.div`
  margin: 1em;
`;

class Dice extends Component {
  render() {
    return (
      <StyledWrapper className="Dice">
        {this.props.dice.map((d, idx) => (
          <Die
            handleClick={this.props.handleClick}
            val={d}
            locked={this.props.locked[idx]}
            idx={idx}
            key={idx}
          />
        ))}
      </StyledWrapper>
    );
  }
}

export default Dice;
