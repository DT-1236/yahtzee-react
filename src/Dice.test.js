import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Dice from './Dice';

const diceTestRender = (
  <Dice dice={[1, 1, 1]} locked={[false, true, false]} handleClick={jest.fn} />
);

describe('simple smoke tests', () => {
  it('renders without crashing', () => {
    shallow(diceTestRender);
  });

  it('mounts without crashing', () => {
    mount(diceTestRender);
  });
});

describe('simple snapshots', () => {
  it('matches the shallow snapshot', () => {
    const wrapper = shallow(diceTestRender);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
