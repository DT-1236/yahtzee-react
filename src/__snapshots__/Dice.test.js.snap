// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple snapshots matches the shallow snapshot 1`] = `
<div
  className="Dice"
>
  <Die
    handleClick={[Function]}
    idx={0}
    key="0"
    locked={false}
    val={1}
  />
  <Die
    handleClick={[Function]}
    idx={1}
    key="1"
    locked={true}
    val={1}
  />
  <Die
    handleClick={[Function]}
    idx={2}
    key="2"
    locked={false}
    val={1}
  />
</div>
`;
